common/sgx/tcbinfo.c

```diff
@@ -327,21 +327,33 @@ static void _determine_platform_tcb_info_tcb_level(
     if (platform_tcb_level->status.AsUINT32 != OE_TCB_LEVEL_STATUS_UNKNOWN)
         return;
 
+    bool comp_svn_out_of_date = false;
+    bool pce_svn_out_of_date = false;
+
     // Compare all of the platform's comp svn values with the corresponding
     // values in the current tcb level.
     for (uint32_t i = 0; i < OE_COUNTOF(platform_tcb_level->sgx_tcb_comp_svn);
          ++i)
     {
         if (platform_tcb_level->sgx_tcb_comp_svn[i] <
             tcb_level->sgx_tcb_comp_svn[i])
-            return;
+            comp_svn_out_of_date = true;
     }
     if (platform_tcb_level->pce_svn < tcb_level->pce_svn)
+        pce_svn_out_of_date = true;
+
+    if (tcb_level->status.fields.up_to_date == 1)
```

Comment:
Can you add a comment explaining why this check is necessary?  It's not obvious given the assumption of `tcb_level.status` is monotonically increasing on iterative calls to this function (as stated in the function comment). If `platform_tcb_level` previously failed to to clear the lower status levels, why don't we set the additional debug bits at the lower tiers of failure as well? (i.e. what is the condition under which the comparison against a higher tier `tcb_level` conflicts with something previously set by a failure at a lower tier?)